src/aegis_ai_web/src/endpoints/kpi.py

```diff
@@ -0,0 +1,141 @@
+    """
+    # Read env var dynamically to support test fixtures that set it
+    log_file = os.getenv(
+        "AEGIS_WEB_FEEDBACK_LOG", f"{get_settings().config_dir}/feedback.csv"
+    )
+    log_path = Path(log_file)
+
+    # Return empty list if file doesn't exist
+    if not log_path.exists():
+        return []
+
+    entries = []
+
+    # Open file with read lock for thread-safe reads
+    with open(log_path, "r", newline="", encoding="utf-8") as csvfile:
+        # Acquire shared lock for thread-safe reads
+        fcntl.flock(csvfile.fileno(), fcntl.LOCK_SH)
+        try:
+            reader = csv.DictReader(csvfile)
+            for row in reader:
+                # Validate entry matches schema
+                if FEEDBACK_SCHEMA.validate_parsed_log(row):
+                    # Normalize accept field to lowercase
+                    if "accept" in row and row["accept"]:
+                        row["accept"] = row["accept"].lower()
+                    entries.append(row)
+        finally:
+            # Release lock
+            fcntl.flock(csvfile.fileno(), fcntl.LOCK_UN)
+
+    return entries
+
+
+class SortOrder(str, Enum):
+    """Sort order for datetime field."""
+
+    ASC = "asc"
+    DESC = "desc"
+
+
+def get_cve_kpi(feature: str, order: SortOrder = SortOrder.ASC) -> KPIScoreResponse:
+    """
+    Get KPI metrics for CVE analysis feedback filtered by feature.
+
+    Args:
+        feature: Feature name to filter entries by
+        order: Sort order for datetime field (default: ASC)
+
+    Returns:
+        KPIScoreResponse with score and entries
+    """
+    try:
+        # Read all log entries
+        all_entries = read_feedback_logs()
+
+        # Filter entries by feature
+        filtered_entries: List[Dict[str, Any]] = [
+            entry.copy() for entry in all_entries if entry.get("feature") == feature
+        ]
+
+        if not filtered_entries:
+            return KPIScoreResponse(
+                score="0%",
+                entries=[],
+            )
+
+        # Sort entries by datetime
+        def parse_datetime(entry: Dict[str, Any]) -> datetime:
+            """Parse datetime string to datetime object for sorting."""
+            dt_str = entry.get("datetime", "")
+            try:
+                # Format: "YYYY-MM-DD HH:MM:SS.mmm"
+                return datetime.strptime(dt_str, "%Y-%m-%d %H:%M:%S.%f")
+            except ValueError:
+                # Fallback for entries without milliseconds
+                try:
+                    return datetime.strptime(dt_str, "%Y-%m-%d %H:%M:%S")
+                except ValueError:
+                    # Return epoch if parsing fails
+                    return datetime.fromtimestamp(0)
+
+        filtered_entries.sort(
+            key=parse_datetime,
+            reverse=(order == SortOrder.DESC),
+        )
+
+        # TODO: Add version information to entries based on git history
```

Comment:
What does `git history` refers to?  The Aegis source code git repo?

Version string of Aegis, which is based on git revision in git snapshots, is available via `aegis_ai.__version__`.

Reply:
The purpose here was to indicate/hint that an indexed mapping of date ranges to Aegis versions could be added here, but I was thinking of just using the client logic to build that index and use the date ranges from this API to obtain the Aegis version